cfgov/v1/models/banners.py

```diff
@@ -0,0 +1,38 @@
+from django.core.validators import RegexValidator
+from django.db import models
+from django.utils.safestring import mark_safe
+
+from wagtail.admin.edit_handlers import FieldPanel, StreamFieldPanel
+from wagtail.core.fields import StreamField
+
+from v1.atomic_elements.molecules import Notification
+
+
+class Banner(models.Model):
+    title = models.CharField(
+        max_length=255,
+        help_text='For internal reference only; does not appear on the site.'
+    )
+    url_patterns = models.CharField(
+        max_length=1000,
+        verbose_name='URL patterns',
+        help_text=mark_safe('A regular expression pattern for matching URLs '
+                            'that should show the banner, for example: '
+                            '<code>contact-us|^/complaint/$</code>'),
+        validators=[
+            RegexValidator(regex=r'[A-Za-z0-9\-_.:/?&|\^$]')
```

Comment:
So this is a regex that validates another regex?

Reply:
Yep. It's regexes all the way down :)

I just entered the possible URL-safe characters off-the-cuff. Notice any I'm missing?